components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/mgt/impl/MonitoringManagerImpl.java

```diff
@@ -214,21 +214,22 @@ public void addMonitoringOperation(List<Device> devices) throws PolicyCompliance
 
             Map<Integer, ComplianceData> tempMap = new HashMap<>();
 
+            if (complianceDatas != null) {
+                for (ComplianceData complianceData : complianceDatas) {
 
-            for (ComplianceData complianceData : complianceDatas) {
+                    tempMap.put(complianceData.getDeviceId(), complianceData);
 
-                tempMap.put(complianceData.getDeviceId(), complianceData);
-
-                if (complianceData.getAttempts() == 0) {
-                    deviceIdsToAddOperation.put(complianceData.getDeviceId(),
-                            deviceIds.get(complianceData.getDeviceId()));
-                } else {
-                    deviceIdsWithExistingOperation.put(complianceData.getDeviceId(),
-                            deviceIds.get(complianceData.getDeviceId()));
-                }
-                if (complianceData.getAttempts() >= 20) {
-                    inactiveDeviceIds.put(complianceData.getDeviceId(),
-                            deviceIds.get(complianceData.getDeviceId()));
+                    if (complianceData.getAttempts() == 0) {
+                        deviceIdsToAddOperation.put(complianceData.getDeviceId(),
+                                                    deviceIds.get(complianceData.getDeviceId()));
+                    } else {
+                        deviceIdsWithExistingOperation.put(complianceData.getDeviceId(),
+                                                           deviceIds.get(complianceData.getDeviceId()));
+                    }
+                    if (complianceData.getAttempts() >= 20) {
+                        inactiveDeviceIds.put(complianceData.getDeviceId(),
+                                              deviceIds.get(complianceData.getDeviceId()));
+                    }
```

Comment:
We can combine these 3 conditions into one if..else block rather than 2 separate.